File: solutions/0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_set<int> s;
        int first = 0;
        int second = 0;

        for(int i=0;i<nums.size();i++){
            if(s.count(target-nums[i])){
                second = i;
                break;
            }
            else{
                s.insert(nums[i]);
            }
        }

        for(int i=0;i<nums.size();i++){
            if(nums[i]==target-nums[second]){
                first = i;
                break;
            }
        }

        return {first,second};

    }
};

File: solutions/0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}
        for i,n in enumerate(nums):
            t = target-n
            if t in m:
                return [m[t],i]
            else:
                m[n]=i
        return [-1,-1]
            
                
        
